agenta-backend/agenta_backend/migrations/postgres/env.py

```diff
@@ -0,0 +1,92 @@
+import asyncio
+from logging.config import fileConfig
+
+from sqlalchemy import pool
+from sqlalchemy.engine import Connection
+from sqlalchemy.ext.asyncio import async_engine_from_config
+
+from alembic import context
+
+from agenta_backend.models.db.shared import db_engine
+
+
+# this is the Alembic Config object, which provides
+# access to the values within the .ini file in use.
+config = context.config
+config.set_main_option("sqlalchemy.url", db_engine.postgres_uri)  # type: ignore
+
+
+# Interpret the config file for Python logging.
+# This line sets up loggers basically.
+if config.config_file_name is not None:
+    fileConfig(config.config_file_name)
+
+# add your model's MetaData object here
+# for 'autogenerate' support
+from agenta_backend.models.db_models import Base  # noqa: F403
+
+# from myapp import mymodel
+# target_metadata = mymodel.Base.metadata
+target_metadata = Base.metadata
+
+# other values from the config, defined by the needs of env.py,
+# can be acquired:
+# my_important_option = config.get_main_option("my_important_option")
+# ... etc.
+
+
+def run_migrations_offline() -> None:
+    """Run migrations in 'offline' mode.
+    This configures the context with just a URL
+    and not an Engine, though an Engine is acceptable
+    here as well.  By skipping the Engine creation
+    we don't even need a DBAPI to be available.
+    Calls to context.execute() here emit the given string to the
+    script output.
+    """
+    url = config.get_main_option("sqlalchemy.url")
```

Comment:
should this be a a real url ? if yes does it make sense to have it as an os env ?

Reply:
Yes, it is a real URL. No, it does not. Because:

1. We shouldn't replace the utility function (`get_main_option`) from the Alembic config object with an OS environment variable.
2. The correct behaviour is to simply [update](https://github.com/Agenta-AI/agenta/pull/1856/files#diff-060392092847b9aff977eeb75b9535f639ed63191f5a902e232ca2bedc436cf0R16) the `sqlalchemy.url` option in the config using the environment variable. This makes things straightforward and allows for future extensions due to Alembic's modular nature.